system/Router/RouteCollection.php

```diff
@@ -1040,12 +1040,14 @@ public function isFiltered(string $search, ?string $verb = null): bool
      *    'role:admin,manager'
      *
      * has a filter of "role", with parameters of ['admin', 'manager'].
+     *
+     * @retun string|string[]
      */
-    public function getFilterForRoute(string $search, ?string $verb = null): string
+    public function getFilterForRoute(string $search, ?string $verb = null)
```

Comment:
This is a breaking API change. But it seems we must change the return type for multiple filters.

Reply:
What are your suggestions?

Reply:
@MGatner This PR contains public API changes. 
Can we include this on the next release or we need to wait for the next major version (v5)?

If we can change the return type, `array`(`string[]`) is better.

Reply:
We have to stick to semantic versioning, which means we need to leave the signature or save this for version 5. Workarounds could be adding a new method and deprecating this one, or returning an encoded string (like a CSV of classes & aliases).

Reply:
If we want to include this in the next version (not v5), 
it seems returning an encoded string (like a CSV of classes & aliases) is the only choice.

@iRedds What do you think?

Reply:
I think this is counterproductive.
The filter alias already contains ":" and "," to separate the alias and parameters. And adding one more separator will complicate read/write (by human). 

An array of strings is the best solution in my opinion.

Reply:
@iRedds I didn't say about `RouteCollection::add()` usage.
```php
$collection->add('foo', 'TestController::foo', ['filter' => ['filter1', 'filter2:param']]);
```
This is not a breaking change. Because the 3rd param type is `?array`.

But if we change public method signatures, it is breaking change.
In this PR, `getFilterForRoute()` and `enableFilter()` are changed.
So this PR will not be able to be merged until the next major version 5.0.

Reply:
I think maintainers are too conservative about breaking changes.
If the backward compatibility of commonly used functions is broken, that's BC. It's my opinion.

The current policy negatively affects the improvement and popularization of the framework.
And version 5.0 looks like a meme "ten thousand years later" (c) SBSP

I have nothing to say. ggwp

Reply:
I also feel strongly constrained by the inability to change the method signatures.

I also understand that the policy of following semantic versioning has its value.